native-android/src/lib/notifications.ts

```diff
@@ -0,0 +1,155 @@
+import * as Device from 'expo-device';
+import * as Notifications from 'expo-notifications';
+import { Platform } from 'react-native';
+import WebView from 'react-native-webview';
+import useHarkState from '../state/hark';
+import useStore from '../state/store';
+import storage from './storage';
+import { NOTIFY_PROVIDER, NOTIFY_SERVICE } from '../constants';
+
+const getHasRequestedNotifications = async () => {
+  try {
+    const value = await storage.load<boolean>({
+      key: 'hasRequestedNotifications',
+    });
+    return value;
+  } catch {
+    return false;
+  }
+};
+
+const setHasRequestedNotifications = (value: boolean) => {
+  storage.save({ key: 'hasRequestedNotifications', data: value });
+};
+
+export const initializePushNotifications = async () => {
+  if (!Device.isDevice) {
+    return false;
+  }
+
+  Notifications.setNotificationHandler({
+    handleNotification: async () => ({
+      shouldShowAlert: true,
+      shouldPlaySound: true,
+      shouldSetBadge: false,
+    }),
+  });
+
+  await Notifications.setNotificationCategoryAsync('message', [
+    {
+      identifier: 'reply',
+      buttonTitle: 'Reply',
+      options: {
+        opensAppToForeground: true,
+      },
+    },
+    {
+      identifier: 'dismiss',
+      buttonTitle: 'Dismiss',
+      options: {
+        opensAppToForeground: false,
+      },
+    },
+  ]);
+
+  if (Platform.OS === 'android') {
+    await Notifications.setNotificationChannelAsync('default', {
+      name: 'default',
+      importance: Notifications.AndroidImportance.MAX,
+      vibrationPattern: [0, 250, 250, 250],
+      lightColor: '#FF231F7C',
```

Comment:
do we need a dark color here?

Reply:
`darkColor` does not appear to be an option:

```
export interface NotificationChannel {
  id: string;
  name: string | null;
  importance: AndroidImportance;
  bypassDnd: boolean;
  description: string | null;
  groupId?: string | null;
  lightColor: string;
  lockscreenVisibility: AndroidNotificationVisibility;
  showBadge: boolean;
  sound: 'default' | 'custom' | null;
  audioAttributes: AudioAttributes;
  vibrationPattern: number[] | null;
  enableLights: boolean;
  enableVibrate: boolean;
}
```

Reply:
ah gotcha didn't know just usually there's a complement haha